kernel/posix/pthread.c

```diff
@@ -0,0 +1,261 @@
+ */
+static pthread_attr_t initPthreadAttrs = {
+	K_LOWEST_THREAD_PRIO,   /* priority */
+	NULL,                   /* stack */
+	K_LOWEST_STACK_SIZE,    /* stack size */
+	THREAD_CONFIG_OPTION,   /* Thread Flags */
+	K_NO_WAIT               /* No Delay in thread start */
+};
+
+static int is_prio_valid(int priority)
+{
+	if ((priority > K_LOWEST_THREAD_PRIO)
+	    || (priority < K_HIGHEST_THREAD_PRIO)) {
+		/* Invalid Prioritty */
+		return EINVAL;
+	} else       {
+		return 0;
+	}
+
+}
+/*
+ *  ======== pthread_attr_init ========
+ *  Initialize Thread Attributes to default Attributes
+ */
+int pthread_attr_init(pthread_attr_t *attr)
+{
+	*attr = initPthreadAttrs;
+	return 0;
+}
+
+/*
+ *  ======== pthread_attr_destroy ========
+ */
+int pthread_attr_destroy(pthread_attr_t *attr)
+{
+	return 0;
+}
+
+/*
+ *  ======== pthread_attr_getschedparam ========
+ *  Get Thread Priority
+ */
+int pthread_attr_getschedparam(const pthread_attr_t *attr,
+			       struct sched_param *schedparam)
+{
+	schedparam->priority = attr->priority;
+	return 0;
+}
+
+
+/*
+ *  ======== pthread_attr_getstack ========
+ *  Get Stack Size and Stack Pointer
+ */
+int pthread_attr_getstack(const pthread_attr_t *attr, void **stackaddr,
+			  size_t *stacksize)
+{
+	*stackaddr = attr->stack;
+	*stacksize = attr->stacksize;
+	return 0;
+}
+
+/*
+ *  ======== pthread_attr_getstacksize ========
+ *  Get Stack Size
+ */
+int pthread_attr_getstacksize(const pthread_attr_t *attr, size_t *stacksize)
+{
+	*stacksize = attr->stacksize;
+	return 0;
+}
+
+
+/*
+ *  ======== pthread_attr_setschedparam ========
+ *  Modify priority inside Thread Attribute
+ */
+int pthread_attr_setschedparam(pthread_attr_t *attr,
+			       const struct sched_param *schedparam)
+{
+	int priority = schedparam->priority;
+
+	/*
+	 *  Task priorities can range from  K_LOWEST_THREAD_PRIO to
+	 *  K_HIGHEST_THREAD_PRIO.
+	 */
+	if (is_prio_valid(priority) != 0) {
+		/* Invalid Prioritty */
+		return EINVAL;
+	}
+	attr->priority = priority;
+	return 0;
+}
+
+/*
+ *  ======== pthread_attr_setstacksize ========
+ */
+
+int pthread_attr_setstack(pthread_attr_t *attr, void *stackaddr,
+			  size_t stacksize)
+{
+
+	if ((stackaddr == NULL) || (stacksize <= K_LOWEST_STACK_SIZE)) {
+		return EINVAL;
+	}
+
+	attr->stack = stackaddr;
+	attr->stacksize = stacksize;
+	return 0;
+}
+
+
+
+int pthread_attr_setstacksize(pthread_attr_t *attr, size_t stacksize)
+{
+	/* Do we need to have stacksize Check */
+	attr->stacksize = stacksize;
+	return 0;
+}
+
+void zephyr_thread_wrapper(void *arg1, void *arg2, void *arg3)
+{
+	void * (*fun_ptr)(void *);
+
+	fun_ptr = arg3;
+	(*fun_ptr)((void *)arg1);
+}
+
+int pthread_create(pthread_t *newthread, const pthread_attr_t *attr,
+		   void *(*threadroutine)(void *), void *arg)
+{
+	static int thread_count;
+
+	if (attr->stack == NULL) {
+		return EINVAL;
+	}
+	if (thread_count < CONFIG_MAX_PTHREAD_COUNT) {
+		k_thread_create(&threads[thread_count++], attr->stack,
```

Comment:
`thread_count++` isn't atomic. Please use the the atomic operation primitives in `atomic.h`.

Also, `newthread` isn't used here.  Shouldn't it be defined in terms of a `k_thread` and that used instead?  Otherwise, things operating on `pthread_t` won't work correctly.